dotcom-rendering/src/components/marketing/epics/ContributionsEpicNewsletterSignup.tsx

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file
+ * This file was migrated from:
+ * https://github.com/guardian/support-dotcom-components/blob/a482b35a25ca59f66501c4de02de817046206298/packages/modules/src/modules/epics/NewsletterSignup.tsx
+ */
+import { css } from '@emotion/react';
+import { isObject, log } from '@guardian/libs';
+import { space } from '@guardian/source-foundations';
+import { useEffect, useRef, useState } from 'react';
+
+const containerStyles = css`
+	margin: ${space[6]}px ${space[2]}px ${space[1]}px 0;
+	width: 100%;
+`;
+
+export const ContributionsEpicNewsletterSignup = ({
+	url,
+}: {
+	url: string;
+}): JSX.Element => {
+	const [iframeHeight, setIframeHeight] = useState(60);
+	const iframeRef = useRef<HTMLIFrameElement>(null);
```

Comment:
[@arelra made me aware that it’s probably best to `useCallback` for this](https://github.com/guardian/dotcom-rendering/pull/9989), and make sure your `useEffect` has `iframeRef` as part of its dependencies array.

Reply:
Nice, I think I'll do this in a follow up PR though